src/sys/windows.rs

```diff
@@ -386,6 +386,64 @@ impl Socket {
         }
     }
 
+    pub fn recv_vectored(&self, bufs: &mut [IoSliceMut<'_>], flags: c_int) -> io::Result<usize> {
+        unsafe {
+            let buffer_count = bufs.len().min(DWORD::MAX as usize) as DWORD;
+            let mut nread = 0;
+            let mut flags = flags as DWORD;
+
+            let ret = sock::WSARecv(
+                self.socket,
+                bufs.as_mut_ptr() as *mut WSABUF,
+                buffer_count,
+                &mut nread,
+                &mut flags,
+                ptr::null_mut(),
+                None,
+            );
+
+            match ret {
+                0 => Ok(nread as usize),
+                _ if sock::WSAGetLastError() == sock::WSAESHUTDOWN => Ok(0),
```

Comment:
Why is the this error being ignored? Sending on a socket that is shutdown should be an error.

Reply:
Yes, but when reading on a socket, unix platforms report this error as a zero-sized read. This is also done in the other `recv` functions.